### PR TITLE
Add release notes and rule updates for september 2022

### PR DESCRIPTION
**Pull request by adamkvitek**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -10,6 +10,69 @@ toc_max_heading_level: 2
...
+
+- Previously, you could only add tags for specific projects in the Semgrep App on the configuration page of a project. Now, you can add tags to the `semgrep` YAML configuration file that is automatically assigned to a project connected with Semgrep App. To manage tags, go to the Semgrep App [Projects](https://semgrep.dev/orgs/-/projects) tab, click on the project's <i class="fa-solid fa-gear"></i> cogwheel icon to enter project configuration page, and then remove or add tags, or add tags to your `semgrep.yml`. However, removing tags added through the `semgrep.yml` does not remove the tag from Semgrep App. To remove these tags, remove them on the project configuration page of the Semgrep App.
+    Example of tags added to `semgrep.yml` file:
+    ```yaml
+    tags:
+        - favourite
+        - awesomeproject
+    ```
```
[the remark this answers is not in the document]
Comment (zyannes) [reply docs/release-notes.md:44]: The `semgrep.yml` should be `.semgrepconfig.yml`.

File: docs/release-notes.md [hunk 2]
```diff
@@ -10,6 +10,73 @@ toc_max_heading_level: 2
...
+
+    Example of tags added to `.semgrepconfig.yml` file:
+    ```yaml
+    tags:
+        - favourite
+        - awesomeproject
+    ```
+    These tags are displayed in Semgrep App [Projects](https://semgrep.dev/orgs/-/projects) tab also.
+
+    However, removing tags added through the `semgrepconfig.yml` does not remove the tag from Semgrep App. To remove these tags, remove them on the project configuration page of the Semgrep App.
```
Comment (adamkvitek) [inline docs/release-notes.md:49]: @zyannes updated. :-)

Comment (zyannes) [reply docs/release-notes.md:49]: Line 49 has `semgrepconfig.yml`, can we change it to `.semgrepconfig.yml`? Besides that, looks good to me!

Comment (adamkvitek) [reply docs/release-notes.md:49]:
```suggestion
- You can add tags for specific projects in the Semgrep App on the configuration page of a project. To manage tags, go to the Semgrep App [Projects](https://semgrep.dev/orgs/-/projects) tab, click on the project's <i class="fa-solid fa-gear"></i> cogwheel icon to enter project configuration page, and then add or remove tags.
    With this update, you can create `.semgrepconfig.yml` file in the root directory of your repository and add tags in this file also.

    Example of tags added to `.semgrepconfig.yml` file:
    ```yaml
    tags:
        - favourite
        - awesomeproject
    ```
    These tags are displayed in Semgrep App [Projects](https://semgrep.dev/orgs/-/projects) tab also.

    However, removing tags added through the `.semgrepconfig.yml` does not remove the tag from Semgrep App. To remove these tags, remove them on the project configuration page of the Semgrep App.
```

**Review by pabloest — APPROVED** [review]

File: docs/release-notes.md [hunk 3]
```diff
@@ -10,6 +10,73 @@ toc_max_heading_level: 2
...
+
+- The Semgrep CLI output now displays non-blocking and blocking findings separately. CLI output also provides a list of the blocking rules that matched the code.
+
+- taint-mode: Experimental support for basic field-sensitive taint tracking. Semgrep can now track `x.a` and `x.b` separately, so that for example: `x.a` can be tainted at the same time as `x.b` is clean, hence `sink(x.a)` can produce a finding but `sink(x.b)` does not. It is also possible for `x` to be tainted while `x.a` is clean. As a result, the number of false positives that Semgrep reports is reduced.
+
+#### Changes
+
+- generic mode: Allow text input without human-readable indentation up to 500 bytes. This value is subject to change. This relaxation is intended to facilitate testing a long line without a trailing newline. Semgrep users should not expect files that are not human-readable to be processed by Semgrep's generic mode, or in any mode for the matter. (Issues [6071](https://github.com/returntocorp/semgrep/issues/6071,) [6162]https://github.com/returntocorp/semgrep/issues/6162)
```
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/release-notes.md:57]:
```suggestion
- generic mode: Allow text input without human-readable indentation up to 500 bytes. This value is subject to change. This relaxation is intended to facilitate testing a long line without a trailing newline. Semgrep users should not expect files that are not human-readable to be processed by Semgrep's generic mode, or in any mode for the matter. (Issues [6071](https://github.com/returntocorp/semgrep/issues/6071), [6162](https://github.com/returntocorp/semgrep/issues/6162)
```

**Pull request merged**